### PR TITLE
Refactoring TemplateCollection to allow for one template.yaml per dataset

### PR DESCRIPTION
**Pull request by arnaudstiegler**
This PR introduces the following changes for #29 
- there is now one `templates.yaml` per dataset/subset
- New class: DatasetTemplates that wraps all templates for a specific dataset, and implements all helper functions to read/write on the template file
- TemplateCollection now initializes all DatasetTemplates and store them in a dictionary
- New `Templates` folder, folder path is `Templates/{dataset_name}/{subset_name}/templates.yaml`
- Necessary updates to `promptsource.py` to account for the updates

I've tested the changes as much as possible (additions, deletions, updates etc.. on the templates), will do a bit more testing tomorrow.

### CODE REVIEW

**Review by VictorSanh — CHANGES_REQUESTED** [review]
Thanks @arnaudstiegler !
LGTM modulo minor comment I left
Let me know once you finished your tests, I'll merge it!

**Comment by VictorSanh** [comment]
@arnaudstiegler I guess it also makes sense to rm the current `templates.yaml` at the root its content is distributed into the `templates` folder

File: promptsource/templates.py [hunk 1]
```diff
@@ -1,104 +1,191 @@
...
+        if template_name not in self.templates.keys():
+            raise ValueError(f"No template with name {template_name} for dataset {self.dataset_name} exists.")
 
-        del self.templates[dataset][template_name]
-        if len(self.templates[dataset]) == 0:
-            del self.templates[dataset]
+        del self.templates[template_name]
 
-    def get_templates(self, dataset):
+        self.write_to_file()
```
[the remark this answers is not in the document]
Comment (arnaudstiegler) [reply promptsource/templates.py]: added

**Comment by arnaudstiegler** [comment]
Added the folder deletion if no template is left.

Tested on as much scenarios as possible, but I expect some issues with corner cases to arise when people play with it again

**Review by VictorSanh — APPROVED** [review]

**Comment by VictorSanh** [comment]
lgtm, thank you!

**Pull request merged**